bluesky/scans.py

```diff
@@ -782,21 +806,22 @@ def args(self):
         # Do this so that args is not settable. Too complex to allow updates.
         return self._args
 
-    def _pre_scan(self):
+    @property
+    def cycler(self):
         # Build a Cycler for ScanND.
-        num = self.num
-        self.cycler = None
+        result = None
         for motor, start, stop, in chunked(self.args, 3):
-            offset = self._offsets[motor]
-            steps = offset + np.linspace(start, stop, num=num, endpoint=True)
+            init_pos = self.init_pos[motor]
+            steps = init_pos  + np.linspace(start, stop, num=self.num,
+                                            endpoint=True)
             c = cycler(motor, steps)
             # Special case first pass because their is no
             # mutliplicative identity for cyclers.
-            if self.cycler is None:
-                self.cycler = c
+            if result is None:
+                result = c
             else:
-                self.cycler += c
-        yield from super()._pre_scan()
+                result += c
```

Comment:
Probably better to make a list of cyclers and then run the list through a reduce.